src/sentry/tagstore/v2/backend.py

```diff
@@ -298,16 +310,349 @@ def delete_tag_key(self, project_id, key):
+        # reasonable matches
+
+        # get initial matches to start the filter
+        k, v = tag_lookups.pop()
+        matches = list(
+            EventTag.objects.filter(
+                project_id=project_id,
+                group_id=group_id,
+                key_id=k,
+                value_id=v,
+                **self._get_environment_filter(environment_id)
+            ).values_list('event_id', flat=True)[:1000]
+        )
+
+        # for each remaining tag, find matches contained in our
+        # existing set, pruning it down each iteration
+        for k, v in tag_lookups:
+            matches = list(
+                EventTag.objects.filter(
+                    project_id=project_id,
+                    group_id=group_id,
+                    event_id__in=matches,
+                    key_id=k,
+                    value_id=v,
+                    **self._get_environment_filter(environment_id)
+                ).values_list('event_id', flat=True)[:1000]
+            )
+            if not matches:
+                return []
+
+        return matches
+
+    def get_groups_user_counts(self, project_id, group_ids, environment_id):
+        qs = GroupTagKey.objects.filter(
+            project_id=project_id,
+            group_id__in=group_ids,
+            _key__key='sentry:user',
+            **self._get_environment_filter(environment_id)
+        )
+
+        return defaultdict(int, qs.values_list('group_id', 'values_seen'))
+
+    def get_group_tag_value_count(self, project_id, group_id, environment_id, key):
+        if db.is_postgres():
+            # This doesnt guarantee percentage is accurate, but it does ensure
+            # that the query has a maximum cost
+            using = router.db_for_read(GroupTagValue)
+            cursor = connections[using].cursor()
+            cursor.execute(
+                """
+                SELECT SUM(t)
+                FROM (
+                    SELECT times_seen as t
+                    FROM tagstore_grouptagvalue
+                    INNER JOIN tagstore_tagkey
+                    ON (tagstore_grouptagvalue.key_id = tagstore_tagkey.id)
+                    WHERE tagstore_grouptagvalue.group_id = %s
+                    AND tagstore_grouptagvalue.environment_id = %s
+                    AND tagstore_tagkey.key = %s
+                    ORDER BY last_seen DESC
+                    LIMIT 10000
+                ) as a
+            """, [group_id, environment_id, key]
+            )
+            return cursor.fetchone()[0] or 0
+
+        cutoff = timezone.now() - timedelta(days=7)
+        return GroupTagValue.objects.filter(
+            group_id=group_id,
+            _key__key=key,
+            last_seen__gte=cutoff,
+            **self._get_environment_filter(environment_id)
+        ).aggregate(t=Sum('times_seen'))['t']
+
+    def get_top_group_tag_values(self, project_id, group_id, environment_id, key, limit=3):
```

Comment:
I know we haven't talked again about these in a ton of detail, but would be nice if we can revisit tracking top-k again in redis. (not this pass for sure, but it'd be nice to not carry this over if we don't have to since it doesn't work great.)